src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -211,6 +222,8 @@ public Task Stop()
                 Logger.DebugFormat("Stopped {0} receiver", receiver.Id);
             });
 
+            rateLimitLoopCancellationToken.Cancel();
```

Comment:
Would we want to await the completion of the rate limiting task or not? If we do, we would need to guard against it throwing a cancellation exception